src/ccl_power.c

```diff
@@ -1234,101 +1312,23 @@ static void ccl_cosmology_compute_power_emu(ccl_cosmology * cosmo, int * status)
     ccl_cosmology_set_status_message(cosmo, "ccl_power.c: ccl_cosmology_compute_power_emu(): sigma8 is not defined; specify sigma8 instead of A_s\n");
     return;
   }
-
-  // Prepare to run CLASS for linear scales
-  ccl_fill_class_parameters(cosmo,&fc,parser_length, status);
-
-  if (*status != CCL_ERROR_CLASS)
-    ccl_run_class(cosmo, &fc,&pr,&ba,&th,&pt,&tr,&pm,&sp,&nl,&le,&op,init_arr,status);
-
-  if (*status == CCL_ERROR_CLASS) {
-    //printed error message while running CLASS
-    ccl_free_class_structs(cosmo, &ba,&th,&pt,&tr,&pm,&sp,&nl,&le,init_arr,status);
-    return;
-  }
-  if (parser_free(&fc)== _FAILURE_) {
-    *status = CCL_ERROR_CLASS;
-    ccl_cosmology_set_status_message(cosmo, "ccl_power.c: ccl_cosmology_compute_power_class(): Error freeing CLASS parser\n");
-    ccl_free_class_structs(cosmo, &ba,&th,&pt,&tr,&pm,&sp,&nl,&le,init_arr,status);
-    return;
-  }
-
-  //These are the limits of the splining range
-  cosmo->data.k_min_lin=2*exp(sp.ln_k[0]);
-  cosmo->data.k_max_lin=ccl_splines->K_MAX_SPLINE;
-  //CLASS calculations done - now allocate CCL splines
-  double kmin = cosmo->data.k_min_lin;
-  double kmax = ccl_splines->K_MAX_SPLINE;
-  //Compute nk from number of decades and N_K = # k per decade
-  double ndecades = log10(kmax) - log10(kmin);
-  int nk = (int)ceil(ndecades*ccl_splines->N_K);
-  double amin = ccl_splines->A_SPLINE_MINLOG_PK;
+  
+  // Run CLASS for linear scales only
+  cosmology_compute_power_class(cosmo, status, 0);
```

Comment:
I see. Since `cosmology_compute_power_class` is an internal function, maybe we could declare it static at the top of the file so it does not appear in the headers?